src/pages/Admin/MapPreview/MapPreview.tsx

```diff
@@ -171,12 +174,13 @@ const MapPreview: React.FC<MapPreviewProps> = ({
               height: 100,
             }}
             otherIconsStyle={{ opacity: 0.4 }}
-            // lockAspectRatio
+            lockAspectRatio
+            isLoading={isLoading}
           />
         )}
 
-        <S.EditButton onClick={handleEditButton}>
-          {isEditing ? "Save layout" : "Edit layout"}
+        <S.EditButton disabled={isLoading} onClick={handleEditButton}>
+          {isLoading ? "Saving..." : isEditing ? "Save layout" : "Edit layout"}
```

Comment:
Perhaps instead of `isLoading` this could be called `isSaving` or similar? Would make the code easier to follow.

The nested ternary here is.. interesting. I don't necessarily hate it.. but I feel like I should for some reason.

Reply:
Yeah, I feel the same way about the nesting here. I'll figure something out to make it more readable.